drivers/ethernet/eth_mcux.c

```diff
@@ -754,10 +754,11 @@ static void generate_mac(u8_t *mac_addr)
 
 	entropy = sys_rand32_get();
 
+	mac_addr[0] |= 0x02; /* force LAA bit */
+
 	mac_addr[3] = entropy >> 8;
 	mac_addr[4] = entropy >> 16;
-	/* Locally administered, unicast */
-	mac_addr[5] = ((entropy >> 0) & 0xfc) | 0x02;
+	mac_addr[5] = entropy >> 0;
```

Comment:
There is LAA bit set in CONFIG_ETH_MCUX_0_UNIQUE_MAC branch below, which sets it to [5] byte, we should probably fix that too in the same go.